src/main/java/io/vertx/core/cli/CommandLine.java

```diff
@@ -0,0 +1,208 @@
+/*
+ *  Copyright (c) 2011-2013 The original author or authors
+ *  ------------------------------------------------------
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *       The Eclipse Public License is available at
+ *       http://www.eclipse.org/legal/epl-v10.html
+ *
+ *       The Apache License v2.0 is available at
+ *       http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.core.cli;
+
+import io.vertx.codegen.annotations.Fluent;
+import io.vertx.codegen.annotations.GenIgnore;
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.cli.impl.DefaultCommandLine;
+
+import java.util.List;
+
+/**
+ * The {@link CommandLineParser} transforms a CLI (a model) into an {@link CommandLine}. This {@link CommandLine}
+ * has stored the argument and option values. Only  instance of {@link CommandLineParser} should create
+ * objects of this type.
+ *
+ * @author Clement Escoffier <clement@apache.org>
+ */
+@VertxGen
+public interface CommandLine {
+
+  /**
+   * Creates a command line object from the {@link CLI}. This object is intended to be used by
+   * the parser to set the argument and option values.
+   *
+   * @param cli the CLI definition
+   * @return the command line object
+   */
+  static CommandLine create(CLI cli) {
+    return new DefaultCommandLine(cli);
+  }
+
+  /**
+   * Sets the various command line arguments. Method called by {@link CommandLineParser}.
+   *
+   * @param args the arguments
+   * @return the current {@link CommandLine}
+   */
+  @Fluent
+  CommandLine setAllArguments(List<String> args);
```

Comment:
I think this could be removed from the interface as the user does not care about it and the CommandLineParser could make a cast to the impl